File: go/korean.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// KoreanService — V3 Korean-language threat analysis.
type KoreanService struct {
	t *internal.Transport
}

func (s *KoreanService) Analyze(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/korean/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: rust/src/resources/korean.rs
use std::sync::Arc;

use reqwest::Method;
use serde_json::Value;

use crate::error::Result;
use crate::transport::Transport;

pub struct Korean {
    transport: Arc<Transport>,
}

impl Korean {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn analyze(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/korean/analyze", Some(body))
            .await
    }
}
